src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int currentCapacity;
+    private T[] entries;
+    private int size;
+
+    public ArrayList() {
+        entries = (T[]) new Object[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == currentCapacity) {
+            grow();
+        }
+        if (index != size) {
```

Comment:
It will be unless check if change my previus comment

Reply:
I'll look, thanks

Reply:
Well, I see no possibilities to improve my code with this.
To many changes to do just for this.
You may write down your version, if you want.

Reply:
Insted of currentCapacity use entries.length.  if (size == entries.length) {}
block if(index != size) is extra one. Just remove it. Because you will change my previus comment where you should replace add(value, size) on  add(value, size - 1).

Reply:
Try to do that)

Reply:
I've deleted currentCapacity already.
I saw your previous comment and I see no sense in it, it will require changes in another blocks and creation even more lines of code.
If you see some kind of solution - write it completely please and test.